assets/src/edit-story/components/panels/alignment/test/useAlignment.js

```diff
@@ -0,0 +1,592 @@
+
+/**
+ * Internal dependencies
+ */
+import DesignPanel from '../../../inspector/design/designPanel';
+import { usePresubmitHandler } from '../../../form';
+import useAlignment from '../useAlignment';
+
+const DEFAULT_EDITOR_PAGE_WIDTH = 440;
+const DEFAULT_EDITOR_PAGE_HEIGHT = 660;
+
+function renderUseAlignement() {
+  const { result } = renderHook(() => useAlignment());
+
+  return result;
+}
+
+describe('useAlignment', () => {
+  describe('single element', () => {
+    let result;
+    let oElement1;
+    let uElement1;
+    let selectedElements, updatedElementList, boundRect;
+    let presubmitHandler1, presubmitHandler2;
+    let registerSubmitHandler;
+    let onSetProperties;
+    let lastProps;
+
+    function CustomPanel(props) {
+      lastProps = { ...props };
+      usePresubmitHandler(presubmitHandler1, []);
+      usePresubmitHandler(presubmitHandler2, []);
+      return <div />;
+    }
+
+    describe('rotated element', () => {
+      beforeEach(() => {
+        oElement1 = {
+          id: '1',
+          x: 25,
+          y: 114,
+          width: 160,
+          height: 28,
+          rotationAngle: 45,
+          type: 'text',
+        };
+        uElement1 = {
+          id: '1',
+          x: 25,
+          y: 114,
+          width: 160,
+          height: 28,
+          frameX: 39,
+          frameY: 62,
+          frameWidth: 132,
+          frameHeight: 132,
+        };
+        selectedElements = [oElement1];
+        updatedElementList = [uElement1];
+        boundRect = {
+          startX: 0,
+          startY: 0,
+          endX: DEFAULT_EDITOR_PAGE_WIDTH,
+          endY: DEFAULT_EDITOR_PAGE_HEIGHT,
+          width: DEFAULT_EDITOR_PAGE_WIDTH,
+          height: DEFAULT_EDITOR_PAGE_HEIGHT,
+        };
+        onSetProperties = jest.fn();
+        presubmitHandler1 = jest.fn(({ x }) => ({ x: x + 1 }));
+        presubmitHandler2 = jest.fn(({ y }) => ({ y: y + 1 }));
+        registerSubmitHandler = (handler) => {
+          return handler;
+        };
+
+        render(
+          <DesignPanel
+            panelType={CustomPanel}
+            selectedElements={selectedElements}
+            onSetProperties={onSetProperties}
+            registerSubmitHandler={registerSubmitHandler}
+          />
+        );
+        result = renderUseAlignement();
+        const { setUpdatedSelectedElementsWithFrame } = result.current;
+        act(() => setUpdatedSelectedElementsWithFrame(updatedElementList));
+      });
+
+      it('should update element x position to left corner of the page on alignleft button click', () => {
+        const { handleAlign } = result.current;
+
+        const { pushUpdate } = lastProps;
+        reactAct(() => handleAlign('left', boundRect, pushUpdate));
+        const {
+          selectedElements: [updatedElementA],
+        } = lastProps;
+        expect(updatedElementA).toStrictEqual({
+          ...oElement1,
+          x: -14,
+        });
```

Comment:
You can also just use `.toMatchObject({x: -14})` and it'll ignore the unprovided props - makes the test a bit simpler.

Or even `.toHaveProperty('x', -14)`, but that might be a bit too simple.

Just an FYI for next time.

Reply:
Thank you!